### ISSUE TITLE
[Research] Detect Emotion Regulation vs Genuine Experience via LPP and Frontal Theta (Reappraisal Detection)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#53]
## Papers

**Primary:** The neural correlates of emotion processing and reappraisal as reflected in EEG
**Authors:** Various
**Year:** 2024 (published Jan 2025)
**Venue:** Biological Psychology (ScienceDirect)
**URL:** https://www.sciencedirect.com/science/article/pii/S0167876024001715

**Supporting:** Decoded EEG neurofeedback-guided cognitive reappraisal training for emotion regulation (PubMed 39555250, 2024)

**Supporting:** EEG entropy modulation as a biomarker of emotion regulation and resilience (IBRO Neuroreports, 2025)

## Key Contribution

Distinguishes between genuine emotional experience and active emotion regulation (cognitive reappraisal). Key finding: during passive emotional experience, the early LPP (400-1000ms) increases for negative stimuli. During reappraisal, this early LPP does NOT decrease — instead, the parietal LPP sustains longer (through picture offset), indicating active cognitive control. This means the temporal dynamics of frontal-parietal EEG power can distinguish "feeling an emotion" from "regulating an emotion."

The supporting neurofeedback paper found that wavelet energy and differential entropy in high-frequency bands (beta/gamma) are specifically associated with neural plasticity changes during emotion regulation training.

The entropy paper (2025) found that EEG entropy modulation serves as a biomarker of emotion regulation capacity and psychological resilience.

## How It Applies to NeuralDreamWorkshop

Currently, the system cannot distinguish between:
- A user who IS angry (genuine experience)
- A user who FEELS angry but is actively calming down (regulation)
- A user who appears calm but is suppressing anger (suppression)

This distinction matters for:
1. **Meditation feedback** — detecting when someone is successfully regulating vs just sitting still
2. **Stress management** — showing "you're stressed but regulating well" vs "stress is uncontrolled"
3. **Session quality** — emotion regulation during focus sessions indicates cognitive engagement
4. **Therapeutic insight** — longitudinal tracking of regulation capacity improvement

### Files to Modify/Create
- `ml/models/emotion_regulation_detector.py` — new detector
- `ml/processing/eeg_processor.py` — add `compute_regulation_markers()` function
- `ml/models/emotion_classifier.py` — add `regulation_state` to emotion output dict

### Core Algorithm
```python
def compute_regulation_markers(signals: np.ndarray, fs: int = 256) -> dict:
    """
    Detect emotion regulation vs genuine experience from 4-channel Muse 2 EEG.
    
    Markers (from 2024 literature):
    1. Frontal theta power increase → cognitive control engagement
    2. Frontal-parietal connectivity shift → top-down regulation
    3. Beta/alpha ratio temporal dynamics → regulation effort
    4. Entropy increase during regulation → more complex processing
    """
    # ch0=TP9, ch1=AF7, ch2=AF8, ch3=TP10
    
    # 1. Frontal theta increase (regulation engages PFC)
    af7_theta = extract_band_powers(signals[1], fs)['theta']
    af8_theta = extract_band_powers(signals[2], fs)['theta']
    frontal_theta = (af7_theta + af8_theta) / 2
    
    # 2. Frontal-temporal theta coherence (top-down control)
    # High coherence = prefrontal engaging temporal regions
    ft_coherence = compute_coherence(signals, fs, band='theta')
    
    # 3. Beta sustained elevation (cognitive effort of reappraisal)
    frontal_beta = (extract_band_powers(signals[1], fs)['beta'] + 
                    extract_band_powers(signals[2], fs)['beta']) / 2
    
    # 4. Entropy change (regulation increases complexity)
    frontal_entropy = (spectral_entropy(signals[1], fs) + 
                       spectral_entropy(signals[2], fs)) / 2
    
    # Regulation index: combines all markers
    # High frontal theta + high coherence + sustained beta + high entropy
    # = active regulation
    regulation_score = np.clip(
        0.35 * frontal_theta_z +      # PFC engagement
        0.25 * ft_coherence_z +        # top-down control
        0.20 * frontal_beta_z +        # cognitive effort
        0.20 * frontal_entropy_z,      # processing complexity
        0, 1
    )
    
    return {
        "regulation_score": float(regulation_score),
        "regulation_state": _classify_regulation(regulation_score),
        "frontal_theta_power": float(frontal_theta),
        "frontal_temporal_coherence": float(ft_coherence),
        "cognitive_effort": float(frontal_beta),
    }

def _classify_regulation(score: float) -> str:
    if score > 0.65:
        return "actively_regulating"   # user is engaging cognitive control
    elif score > 0.35:
        return "mild_regulation"       # some cognitive engagement
    else:
        return "passive_experience"    # genuine unregulated emotion
```

### Integration with Emotion Output
```python
# In emotion_classifier.py _predict_features():
result = {
    "emotion": dominant_emotion,
    "probabilities": probs,
    "valence": valence,
    "arousal": arousal,
    # NEW: emotion regulation context
    "regulation_state": reg_markers["regulation_state"],
    "regulation_score": reg_markers["regulation_score"],
}
```

### UI Display
- During meditation: "Regulation Active" badge when score > 0.65
- Emotion display: "Stressed (but regulating)" vs "Stressed (escalating)"
- Session summary: regulation score trend chart

## Expected Performance Impact

- No direct accuracy benchmark exists for regulation detection on 4-channel consumer EEG
- The underlying markers (frontal theta, coherence) are well-validated in research EEG
- Expected regulation/non-regulation discrimination: ~65-72% with 4-second epochs
- Main value: qualitative insight layer on top of existing emotion detection
- Risk: frontal theta also increases with drowsiness/mind-wandering — need arousal context to disambiguate

## Complexity: MEDIUM

- Requires reliable theta extraction (already have) and coherence computation (already have `compute_coherence()`)
- Need z-score normalization against baseline (BaselineCalibrator already exists)
- Training data for regulation-specific model would require labeled regulation tasks
- Initial implementation can be threshold-based (no training needed)
- Disambiguation from drowsiness requires combining with arousal index
